framework/test/unit/modules/interoperability/sidechain/module.spec.ts

```diff
@@ -0,0 +1,576 @@
+import { utils } from '@liskhq/lisk-cryptography';
+import {
+	CreateGenesisBlockContextParams,
+	InMemoryPrefixedStateDB,
+} from '../../../../../src/testing';
+import { ChainStatus, SidechainInteroperabilityModule } from '../../../../../src';
+import {
+	activeValidator,
+	chainData,
+	chainInfo,
+	chainValidators,
+	createInitGenesisStateContext,
+	genesisInteroperability,
+	lastCertificate,
+	terminatedOutboxAccount,
+	terminatedStateAccount,
+} from '../interopFixtures';
+import { PrefixedStateReadWriter } from '../../../../../src/state_machine/prefixed_state_read_writer';
+import {
+	computeValidatorsHash,
+	getMainchainID,
+	getMainchainTokenID,
+	validNameCharset,
+} from '../../../../../src/modules/interoperability/utils';
+import {
+	CHAIN_NAME_MAINCHAIN,
+	EMPTY_HASH,
+	HASH_LENGTH,
+	MAX_CHAIN_NAME_LENGTH,
+	MIN_CHAIN_NAME_LENGTH,
+} from '../../../../../src/modules/interoperability/constants';
+
+describe('initGenesisState', () => {
+	const chainID = Buffer.from([1, 2, 3, 4]);
+	let params: CreateGenesisBlockContextParams;
+	let stateStore: PrefixedStateReadWriter;
+	let interopMod: SidechainInteroperabilityModule;
+
+	beforeEach(() => {
+		stateStore = new PrefixedStateReadWriter(new InMemoryPrefixedStateDB());
+		interopMod = new SidechainInteroperabilityModule();
+		params = {
+			stateStore,
+			chainID,
+		};
+	});
+
+	describe('when chainInfos is empty', () => {
+		const genesisInteropWithEmptyChainInfos = {
+			...genesisInteroperability,
+			chainInfos: [],
+		};
+		const ifChainInfosIsEmpty = 'if chainInfos is empty.';
+
+		it('should throw error if ownChainName is the not empty string', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...genesisInteropWithEmptyChainInfos,
+					ownChainName: 'xyz',
+				},
+				params,
+			);
+
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				`ownChainName must be empty string, ${ifChainInfosIsEmpty}`,
+			);
+		});
+
+		it('should throw error if ownChainNonce !== 0', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...genesisInteropWithEmptyChainInfos,
+					ownChainName: '',
+					ownChainNonce: BigInt(1),
+				},
+				params,
+			);
+
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				`ownChainNonce must be 0, ${ifChainInfosIsEmpty}.`,
+			);
+		});
+
+		it('should throw error terminatedStateAccounts is not empty', async () => {
+			const context = createInitGenesisStateContext(
+				{
+					...genesisInteropWithEmptyChainInfos,
+					ownChainName: '',
+					ownChainNonce: BigInt(0),
+					terminatedStateAccounts: [
+						{
+							chainID,
+							terminatedStateAccount,
+						},
+					],
+				},
+				params,
+			);
+
+			await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+				`terminatedStateAccounts must be empty, ${ifChainInfosIsEmpty}.`,
+			);
+		});
+	});
+
+	describe('when chainInfos is not empty', () => {
```

Comment:
can be added within describe `_verifyChainInfos` including above describe `when chainInfos is empty`